src/Hydra/Serializer/CollectionNormalizer.php

```diff
@@ -86,11 +86,16 @@ public function normalize($object, $format = null, array $context = [])
         }
 
         $data['@type'] = 'hydra:Collection';
-
         $data['hydra:member'] = [];
-        $iriOnly = $context[self::IRI_ONLY] ?? $this->defaultContext[self::IRI_ONLY];
-        foreach ($object as $obj) {
-            $data['hydra:member'][] = $iriOnly ? ['@id' => $this->iriConverter->getIriFromItem($obj)] : $this->normalizer->normalize($obj, $format, $context);
+
+        if (isset($context[self::IRI_ONLY]) && true === $context[self::IRI_ONLY] ?? $this->defaultContext[self::IRI_ONLY]) {
```

Comment:
```suggestion
$data['hydra:member'][] = $iriOnly ? [$this->iriConverter->getIriFromItem($obj) : $this->normalizer->normalize($obj, $format, $context);
```

This should be enough.

Reply:
I thought it was faster to check once before the foreach rather than inside each loop.